src/test/java/com/jcabi/github/StatusesTest.java

```diff
@@ -0,0 +1,117 @@
+ * modification, are permitted provided that the following conditions
+ * are met: 1) Redistributions of source code must retain the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer. 2) Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution. 3) Neither the name of the jcabi.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.github;
+
+import com.google.common.base.Optional;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link Statuses}.
+ * @author Chris Rebert (github@chrisrebert.com)
+ * @version $Id$
+ * @since 0.24
+ */
+public final class StatusesTest {
+    /**
+     * Test status URL.
+     */
+    private static final String URL = "http://status.jcabi-github.invalid/42";
+
+    /**
+     * Test commit status context string.
+     */
+    private static final String CONTEXT = "jcabi/github/test";
+
+    /**
+     * StatusCreate can convert itself to JSON.
+     */
+    @Test
+    public void convertsToJsonWhenAllPresent() {
+        final String success = "Everything is awesome";
+        MatcherAssert.assertThat(
+            new Statuses.StatusCreate(Status.State.SUCCESS)
+                .withTargetUrl(Optional.of(URL))
+                .withDescription(success)
+                .withContext(Optional.of(CONTEXT))
+                .json().toString(),
+            // @checkstyle LineLength (1 line)
+            Matchers.equalTo("{\"state\":\"success\",\"description\":\"Everything is awesome\",\"context\":\"jcabi/github/test\",\"target_url\":\"http://status.jcabi-github.invalid/42\"}")
+        );
+    }
+
+    /**
+     * StatusCreate can convert itself to JSON when it has no URL.
+     */
+    @Test
+    public void convertsToJsonWhenUrlAbsent() {
+        final String success = "Living the dream!";
```

Comment:
@cvrebert Same here, better to pass it directly to `withDescription` as a literal.